### PR TITLE
Add resource health to `azurerm_monitor_activity_log_alert`

### LINKED ISSUE #9391: Support for granular alert conditon when setting resource health alert with terraform
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Terraform lacks of ability to further set alert conditions for resource health alert

### New or Affected Resource(s)

azurerm_monitor_activity_log_alert

### Potential Terraform Configuration

```hcl
resource "azurerm_monitor_activity_log_alert" "resourcehealth" {
  name                = " Resource Health Alert"
  description         = "Resource Health Alerts"
  resource_group_name = var.resource_group_name
  scopes              = [var.scopes]
  criteria {
    category = "ResourceHealth"
  }
 alert_condition{
   Event_status =["active", "in progress", "resolved","updated" ]
    current_resource_status=["available","degraded","unavailable","unknown"]
    previous_resource_status=["available","degraded","unavailable","unknown"]
    reason_type=["platform initiated", "user initiated"]
}

  action {
    action_group_id = var.action_group_id
  }
  tags = var.tags
}
```

### References

* #0000

### PR DESCRIPTION
**Pull request by tj-spyro**
Add resource health to `azurerm_monitor_activity_log_alert`

Fixes #9391

### CODE REVIEW

File: internal/services/monitor/monitor_activity_log_alert_resource.go [hunk 1]
```diff
@@ -160,6 +160,64 @@ func resourceMonitorActivityLogAlert() *pluginsdk.Resource {
 							ConflictsWith: []string{"criteria.0.recommendation_category", "criteria.0.recommendation_impact"},
 						},
 						//lintignore:XS003
+						"resource_health": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"current_health": {
```
Comment (katbyte) [inline internal/services/monitor/monitor_activity_log_alert_resource.go:169]: we can probably remove health as its part of the block name

File: internal/services/monitor/monitor_activity_log_alert_resource.go [hunk 2]
```diff
@@ -160,6 +160,64 @@ func resourceMonitorActivityLogAlert() *pluginsdk.Resource {
...
+												"Degraded",
+												"Unavailable",
+												"Unknown",
+											},
+												false,
+											),
+										},
+										Set: pluginsdk.HashString,
+									},
+									"previous_health": {
```
Comment (katbyte) [inline internal/services/monitor/monitor_activity_log_alert_resource.go:185]: and here
```suggestion
									},
									
									"previous": {
```

File: internal/services/monitor/monitor_activity_log_alert_resource.go (hunk 1 above)
Comment (tj-spyro) [reply internal/services/monitor/monitor_activity_log_alert_resource.go]: This is resolved

File: internal/services/monitor/monitor_activity_log_alert_resource.go (hunk 2 above)
Comment (tj-spyro) [reply internal/services/monitor/monitor_activity_log_alert_resource.go]: This is resolved

**Comment by tj-spyro** [comment]
@katbyte - comments resolved, ready for another review

File: website/docs/r/monitor_activity_log_alert.html.markdown [hunk 3]
```diff
@@ -96,10 +96,19 @@ A `criteria` block supports the following:
...
 * `recommendation_impact` - (Optional) The recommendation impact of the event. Possible values are `High`, `Medium` and `Low`. It is only allowed when `category` is `Recommendation`.
+* `resource_health` - (Optional) A block to define fine grain resource health settings.
 * `service_health` - (Optional) A block to define fine grain service health settings.
 
 ---
 
+A `resource_health` block supports the following:
+
+* `current` (Optional) Current resource health status this alert will monitor. Possible values are `Available`, `Degraded`, `Unavailable` and `Unknown`.
+* `previous` (Optional) Previous resource health status this alert will monitor. Possible values are `Available`, `Degraded`, `Unavailable` and `Unknown`.
```
Comment (katbyte) [inline website/docs/r/monitor_activity_log_alert.html.markdown:107]: small phrasing suggestion
```suggestion
* `current` (Optional) The current resource health statuses that will log an alert. Possible values are `Available`, `Degraded`, `Unavailable` and `Unknown`.
* `previous` (Optional) The previous resource health statuses that will log an alert. Possible values are `Available`, `Degraded`, `Unavailable` and `Unknown`.
```

File: website/docs/r/monitor_activity_log_alert.html.markdown [hunk 4]
```diff
@@ -96,10 +96,19 @@ A `criteria` block supports the following:
...
+* `reason` (Optional) Reason this alert will monitor. Possible values are `PlatformInitiated`, `UserInitiated` and `Unknown`.
```
Comment (katbyte) [inline website/docs/r/monitor_activity_log_alert.html.markdown:108]: could we phrase this a big better to accurately describe what a reason is and what its monitoring?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @tj-spyro - couple small comments on docs and some test failures:
```
------- Stdout: -------
=== RUN   TestAccMonitorActivityLogAlert_ResourceHealth_basicAndUpdate
=== PAUSE TestAccMonitorActivityLogAlert_ResourceHealth_basicAndUpdate
=== CONT  TestAccMonitorActivityLogAlert_ResourceHealth_basicAndUpdate
    testcase.go:110: Step 1/6 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
        stdout
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_monitor_activity_log_alert.test will be updated in-place
          ~ resource "azurerm_monitor_activity_log_alert" "test" {
                id                  = "/subscriptions/*******/resourceGroups/acctestRG-220202220513688243/providers/Microsoft.Insights/activityLogAlerts/acctestActivityLogAlert-220202220513688243"
                name                = "acctestActivityLogAlert-220202220513688243"
                tags                = {}
                # (4 unchanged attributes hidden)
        
        
              ~ criteria {
                    # (1 unchanged attribute hidden)
        
                  ~ resource_health {
                      ~ current  = [
                          + "Degraded",
                          + "Unavailable",
                        ]
                      ~ previous = [
                          + "Available",
                          + "Unknown",
                        ]
                        # (1 unchanged attribute hidden)
                    }
                }
                # (2 unchanged blocks hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccMonitorActivityLogAlert_ResourceHealth_basicAndUpdate (211.98s)
FAIL
```

**Comment by tj-spyro** [comment]
@katbyte, sorry for the delay in getting the changes made. Should be good now

**Review by katbyte — APPROVED** [review]
Thanks @tj-spyro - LGTM now 🚀

**Pull request merged**